aopy/data.py

```diff
@@ -16,6 +16,35 @@
 import pickle
 import yaml
 from .utils import get_pulse_edge_times, compute_pulse_duty_cycles
+import shutil
+
+def move_files(files_list, dest_dir, overwrite = True):
+    '''
+    This is a function that moves files/folders in files_list to dest_dir. This function is part of a backup script.
+    Args:
+        files_list (list): List of strings with filenames including full filepath
+        dest_dir (str): Destination fullpath
+        overwrite (bool): bool to allow overwriting files in destination directory
+
+    Returns: Nothing
+
+    '''
+    for file_to_move in files_list:
+        if file_to_move:
+            source = file_to_move[0] # assuming there will only ever be one hdf file and one ecube directory per entry, which is the case in our setup..
+            dest = os.path.join(dest_dir, os.path.basename(source))
+            # print(dest)
+            if not os.path.exists(dest):
+                print(f"Copying {dest}...", end="")
+                if os.path.isfile(source):
+                    shutil.copy2(source, dest)
+                else:
+                    shutil.copytree(source,dest, dirs_exist_ok= overwrite)
+                print("")
+            else:
+                print(f"File {source} already moved!")
```

Comment:
i tried using this to backup the files from our optogenetics experiments, i had a couple problems
- the destination directory needs to be the full path, including the system name. if you're copying arbitrary systems, this can be annoying
- if the system folder doesn't exist, it should be created

I came up with the following:

```
def move_files(files_list, source_dir, dest_dir):
    for file in files_list:
        source = os.path.join(source_dir, file)
        dest = os.path.join(dest_dir, file)
        dirname = os.path.dirname(dest)
        if not os.path.exists(dirname):
            os.mkdir(dirname)
        if not os.path.exists(dest):
            print(f"Copying {dest}...", end="")
            if os.path.isfile(source):
                shutil.copy2(source, dest)
            else:
                shutil.copytree(source,dest)
            print("")
        else:
            print(f"File {source} already moved!")
```